homeassistant/util/package.py

```diff
@@ -34,14 +34,24 @@ def is_installed(package: str) -> bool:
     Returns False when the package is not installed or doesn't meet req.
     """
     try:
+        pkg_resources.get_distribution(package)
+        return True
+    except (pkg_resources.ResolutionError, pkg_resources.ExtractionError):
```

Comment:
Why is this needed? I don't fully follow what is going on here. The version check below if it returned none aught to have caused function to return False, since None would not be in req. 

So is this trying to accept the faulty installed package anyway?

Reply:
get_distribution will not trip up on any cruft left behind like importlib.metadata suffers from.  If it is successful and does not throw the package is installed and meets the requirements.

If it thinks the package is not installed we fall back to the previous behavior to preserve compatibility.

My first attempt of this just looked for version returning none and that worked fine but it had the unfortunate side effect of reinstalling the package every restart since nothing would ever clean up the cruft that importlib.metadata was tripped up on.

Reply:
That mean we leave the cruft around forever. Shouldn't we at least warn in that case?